runtime/compiler/env/annotations/Annotations.cpp

```diff
@@ -107,8 +107,8 @@ TR_Debug::printAnnotationInfoEntry(J9AnnotationInfo * annotationInfo,
          signatureName = "";
          if ((flag & ~ANNOTATION_PARM_MASK) == ANNOTATION_TYPE_PARAMETER)
             {
-            sprintf(annNameBuffer,"parm(%d)",(flag & ANNOTATION_PARM_MASK) >> ANNOTATION_PARM_SHIFT);
-            TR_ASSERT( strlen(annNameBuffer) < ANNO_NAMEBUF_LEN, "buffer length somehow exceeded\n");
+            size_t len = snprintf(annNameBuffer, sizeof(annNameBuffer), "parm(%d)",(flag & ANNOTATION_PARM_MASK) >> ANNOTATION_PARM_SHIFT);
+            TR_ASSERT(len < sizeof(annNameBuffer), "buffer length somehow exceeded\n");
```

Comment:
If `TR_ASSERT` doesn't evaluate the condition (see `Assert.hpp`) then `len` will be unused. @dsouzai should we be concerned about that?

Reply:
I don't think so; if there is a warning about it that is treated as an error, probably best to just add a pragma directive to ignore it in this location.

Reply:
Yeah I think that's fine; `TR_ASSERT` is enabled under:
```
#if defined(DEBUG) || defined(PROD_WITH_ASSUMES)
```